p2p/src/lib.rs

```diff
@@ -7,27 +7,44 @@ use std::{io, net::AddrParseError};
 use iroha_crypto::ursa::{
     encryption::symm::prelude::ChaCha20Poly1305, kex::x25519::X25519Sha256, CryptoError,
 };
-pub use network::{ConnectPeer, DisconnectPeer, NetworkBase, Post};
+pub use network::message::*;
 use parity_scale_codec::{Decode, Encode};
 use thiserror::Error;
 
-pub mod handshake;
 pub mod network;
 pub mod peer;
 
 /// The main type to use for secure communication.
-pub type Network<T> = NetworkBase<T, X25519Sha256, ChaCha20Poly1305>;
+pub type NetworkHandle<T> = network::NetworkBaseHandle<T, X25519Sha256, ChaCha20Poly1305>;
+
+pub mod boilerplate {
+    //! Module containing trait shorthands. Remove when trait aliases
+    //! are stable <https://github.com/rust-lang/rust/issues/41517>
+
+    use iroha_crypto::ursa::{encryption::symm::Encryptor, kex::KeyExchangeScheme};
+
+    use super::*;
+
+    /// Shorthand for traits required for payload
+    pub trait Pload: Encode + Decode + Send + Clone + 'static {}
+    impl<T> Pload for T where T: Encode + Decode + Send + Clone + 'static {}
+
+    /// Shorthand for traits required for key exchange
+    pub trait Kex: KeyExchangeScheme + Send + 'static {}
+    impl<T> Kex for T where T: KeyExchangeScheme + Send + 'static {}
+
+    /// Shorthand for traits required for encryptor
+    pub trait Enc: Encryptor + Send + 'static {}
+    impl<T> Enc for T where T: Encryptor + Send + 'static {}
+}
 
 /// Errors used in [`crate`].
-#[derive(Debug, Error, iroha_actor::Message)]
+#[derive(Debug, Error)]
```

Comment:
Food for thought: in most cases the error doc-string and the error description are identical. Maybe a proc-macro replacement for `anyhow` could handle this too?

Reply:
If error message is string literal without formatting it should be possible.

Reply:
And if it's a format string, the part that needs formatting could be concatenated to the doc-string. E.g. 
```rust
/// Too many inputs 
#[error("{inputs}, maximum: {maximum}")]
TooManyInputs
```

Reply:
Created #3374